src/App.tsx

```diff
@@ -21,12 +21,12 @@ const App = observer(() => {
             <div key={task.id}>
               <input type="checkbox"
                 checked={task.completed}
-                onChange={action(e => task.completed = e.target.checked)} />
+                onChange={e => task.completed = e.target.checked} />
               <input
                 value={task.title}
-                onChange={action(e => task.title = e.target.value)} />
+                onChange={e => task.title = e.target.value} />
               <button type="button" onClick={() => store.saveTask(task)}>Save</button>
-              <button type="button" onClick={() => store.deleteTask(task)}>Delete</button>
+              <button type="button" onClick={() => task.delete()}>Delete</button>
```

Comment:
```suggestion
              <button type="button" onClick={task.delete}>Delete</button>
```

Reply:
Not sure - I need to look into this, it could be that in my current implementation I use methods and not const arrow functions - that may break.

Maybe I should fix that  - I'll look into that

Reply:
Oh, yeah, sorry - I just assumed those functions are bound to their context cause that's the only way it is sane to work. I have this entry my `no-restricted-syntax` eslint rule config: 
```js
      {
        // Docs: https://eslint.org/docs/developer-guide/selectors
        // Playground: https://astexplorer.net/#/gist/4d51bf7236eec2a73dcd0cb81d132305/6e36b2ce6f431de9d75620dd5db70fcf5a73f0b9
        selector: 'ClassBody > MethodDefinition[kind=method]',
        message:
          "Methods like `foo() {}` aren't allowed due to dynamic `this` binding. Use lexically bound field initializers instead: `foo = () => {}`.",
      },
```